### PR TITLE
feat: implement -D, -O options for add command

### PR DESCRIPTION
**Pull request by vramana**
One caveat is if a package already exists in another dependecy group, we don't remove the existing entry. Can we fix it in another PR?

### CODE REVIEW

File: crates/package_json/src/lib.rs [hunk 1]
```diff
@@ -16,6 +17,22 @@ pub struct PackageJson {
     value: Value,
 }
 
+pub enum DependencyType {
```
Comment (vramana) [inline crates/package_json/src/lib.rs:20]: Should this be called DependencyGroup instead? Should I add peer dependencies, bundled dependencies in this PR?

File: crates/cli/src/commands.rs [hunk 2]
```diff
@@ -23,8 +24,26 @@ pub enum Subcommands {
...
 }
+
+impl AddArgs {
+    pub fn get_dependency_type(&self) -> DependencyType {
+        if self.dev {
+            DependencyType::Dev
+        } else if self.optional {
+            DependencyType::Optional
+        } else {
+            DependencyType::Normal
```
Comment (anonrig) [inline crates/cli/src/commands.rs:46]: What do you think about naming this as `Default`?

File: crates/package_json/src/lib.rs (hunk 1 above)
Comment (anonrig) [reply crates/package_json/src/lib.rs:20]: I like dependencygroup better, but I'm ok with dependencytype as well. I think we can split the PRs. That's ok

File: crates/package_json/src/lib.rs [hunk 3]
```diff
@@ -16,6 +17,22 @@ pub struct PackageJson {
...
+    Normal,
+    Dev,
+    Optional,
+}
+
+impl Into<&str> for DependencyType {
```
Comment (anonrig) [inline crates/package_json/src/lib.rs:26]: Can you also add From trait as well?

File: crates/cli/src/commands.rs [hunk 4]
```diff
@@ -23,8 +24,26 @@ pub enum Subcommands {
 pub struct AddArgs {
     /// Name of the package
     pub package: String,
+    /// Add the package as a dev dependency
+    #[arg(short = 'D', group = "dependency_type")]
+    dev: bool,
+    /// Add the package as a optional dependency
+    #[arg(short = 'O', group = "dependency_type")]
```
Comment (anonrig) [inline crates/cli/src/commands.rs:31]: long value should be added as `save-optional`

File: crates/cli/src/commands.rs [hunk 5]
```diff
@@ -23,8 +24,26 @@ pub enum Subcommands {
 pub struct AddArgs {
     /// Name of the package
     pub package: String,
+    /// Add the package as a dev dependency
+    #[arg(short = 'D', group = "dependency_type")]
```
Comment (anonrig) [inline crates/cli/src/commands.rs:28]: long value should be added as `save-dev`

**Review by anonrig — APPROVED** [review]

**Comment by vramana** [comment]
> Can you add a TODO in the code so we don't forget?

Sure

**Review by anonrig — APPROVED** [review]

**Comment by anonrig** [comment]
Is this PR ready to get merged?

**Comment by vramana** [comment]
@anonrig Yes!

**Pull request merged**